### PR TITLE
Adopt 2024 11 01 preview

### PR DESCRIPTION
**Pull request by nick863**
# Description

Changes in 2024-11-01-preview:
1. `ApiResponseFormat` renamed to `ResponseFormat`
    ResponseFormat now can be a new object `ResponseFormatJsonSchema`
    this object can be given to CreateAgentOptions, UpdateAgentOptions and in the response it will be in the  ThreadRun, 
2. `AgentOperations.list_run_steps`, `AgentOperations.get_run_step` and `AgentOperations.create_run` will now have a query parameter include with the only supported value \["step_details.tool_calls[*].file_search.results[*].content"\].
3. `RunStepFileSearchToolCall` object will now contain new `RunStepFileSearchToolCallResults` object in `file_search` field, previously it was reserved. `RunStep>RunStepToolCallDetails>RunStepToolCall[]>RunStepFileSearchToolCall>RunStepFileSearchToolCallResults`
4. `VectorStoreFileErrorCode` now can take values "server_error", "invalid_file" and "unsupported_file"  instead of "internal_error", "file_not_found" and "unsupported_file"

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ai/azure-ai-projects/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
 
+## 1.0.0b3 (Unreleased)
+
+### Features Added
+
+* Redesigned streaming and event handlers for agents 
+* Add `parallel_tool_calls` parameter to allow parallel tool execution.
+* Add support for Structured Outputs.
+* Add option to include file contents, when index search is used.
+
```
Comment (dargilco) [inline sdk/ai/azure-ai-projects/CHANGELOG.md:11]: Make sure "for Agents" is mentioned in every line

File: sdk/ai/azure-ai-projects/azure/ai/projects/_version.py [hunk 2]
```diff
@@ -6,4 +6,4 @@
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
 
-VERSION = "1.0.0b2"
+VERSION = "1.0.0b3"
```
Comment (dargilco) [inline sdk/ai/azure-ai-projects/azure/ai/projects/_version.py:9]: This has already been done in the feature branch. Please rebase before sending PR.

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_stream_iteration_with_file_search.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+FILE: sample_agents_stream_iteration_with_file_search.py
```
Comment (dargilco) [inline sdk/ai/azure-ai-projects/samples/agents/sample_agents_stream_iteration_with_file_search.py:7]: Remove "FILE: ..." from all samples

**Review by dargilco — APPROVED** [review]

File: sdk/ai/azure-ai-projects/CHANGELOG.md (hunk 1 above)
Comment (nick863) [reply sdk/ai/azure-ai-projects/CHANGELOG.md]: Fixed.

File: sdk/ai/azure-ai-projects/azure/ai/projects/_patch.py [hunk 4]
```diff
@@ -136,7 +136,7 @@ def __init__(  # pylint: disable=super-init-not-called,too-many-statements
             resource_group_name=resource_group_name,
             project_name=project_name,
             credential=credential,
-            api_version="2024-07-01-preview",  # TODO: Update me
+            api_version="2024-12-01-preview",
             credential_scopes=["https://ml.azure.com/.default"],
```
[the remark this answers is not in the document]
Comment (nick863) [reply sdk/ai/azure-ai-projects/azure/ai/projects/_patch.py]: Reased and now this change is not showing up.

File: sdk/ai/azure-ai-projects/azure/ai/projects/_version.py (hunk 2 above)
Comment (nick863) [reply sdk/ai/azure-ai-projects/azure/ai/projects/_version.py]: Rebased

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_stream_iteration_with_file_search.py (hunk 3 above)
Comment (nick863) [reply sdk/ai/azure-ai-projects/samples/agents/sample_agents_stream_iteration_with_file_search.py]: Fixed.

File: sdk/ai/azure-ai-projects/samples/agents/async_samples/sample_agents_json_schema_async.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+DESCRIPTION:
+    This sample demonstrates how to use agents with JSON schema output format.
+
+USAGE:
+    python sample_agents_json_schema.py
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/samples/agents/async_samples/sample_agents_json_schema_async.py:11]: sample_agents_json_schema_async.py

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_json_schema.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+DESCRIPTION:
+    This sample demonstrates how to use agents with JSON schema output format.
+
+USAGE:
+    python sample_agents_json_schema_async.py
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/samples/agents/sample_agents_json_schema.py:11]: sample_agents_json_schema.py

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_json_schema.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+from azure.ai.projects.models import (
+    MessageTextContent,
+    MessageRole,
+    ResponseFormatJsonSchema,
+    ResponseFormatJsonSchemaType,
+    RunStatus,
+)
+
+# Create an Azure AI Client from a connection string, copied from your AI Studio project.
+# At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/samples/agents/sample_agents_json_schema.py:36]: Should <HubName> be <ProjectName>?

File: sdk/ai/azure-ai-projects/CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,8 @@
 
 ### Features Added
 
+* Add support for Structured Outputs for Agents.
+* Add option to include file contents, when index search is used for Agents.
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/CHANGELOG.md:8]: File contents is included in the response? Will this be used e.g. when doing evaluation for grounding?

**Review by jhakulin — APPROVED** [review]

File: sdk/ai/azure-ai-projects/samples/agents/async_samples/sample_agents_json_schema_async.py (hunk 5 above)
Comment (nick863) [reply sdk/ai/azure-ai-projects/samples/agents/async_samples/sample_agents_json_schema_async.py:11]: fixed.

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_json_schema.py (hunk 7 above)
Comment (nick863) [reply sdk/ai/azure-ai-projects/samples/agents/sample_agents_json_schema.py]: Removed this code as it was removed from all other samples.

File: sdk/ai/azure-ai-projects/samples/agents/sample_agents_json_schema.py (hunk 6 above)
Comment (nick863) [reply sdk/ai/azure-ai-projects/samples/agents/sample_agents_json_schema.py]: Fixed.

File: sdk/ai/azure-ai-projects/CHANGELOG.md (hunk 8 above)
Comment (nick863) [reply sdk/ai/azure-ai-projects/CHANGELOG.md:8]: Yes, evaluation can use this file as a context. Per my understanding user can add this parameter to `data_mapping` parameter of EvaluatorConfiguration.

**Pull request merged**